src/backend/plugins/video/warren_video/api.py

```diff
@@ -26,63 +19,23 @@
 logger = logging.getLogger(__name__)
 
 
-class VideoDayViews(BaseModel):
-    """Model to represent video views for a date."""
-
-    day: Date
-    views: int = 0
-
-
-class VideoViews(BaseModel):
-    """Model to represent video views."""
-
-    total: int
-    daily_views: List[VideoDayViews]
-
-
-@router.get("/{video_id:path}/views")
+@router.get("/{video_uuid:path}/views")
 async def views(
-    video_id: IRI, filters: Annotated[BaseQueryFilters, Depends()]
-) -> VideoViews:
-    """Video views."""
-    query = {
-        "verb": PlayedVerb().id,
-        "activity": video_id,
-        "since": filters.since.isoformat(),
-        "until": filters.until.isoformat(),
-    }
-
+    video_uuid: IRI, filters: Annotated[BaseQueryFilters, Depends()]
+) -> Response[VideoViews]:
+    """Number of views for `video_uuid` in the `since` -> `until` date range."""
+    indicator = DailyVideoViews(
+        client=lrs_client,
+        video_uuid=video_uuid,
+        date_range=DatetimeRange(since=filters.since, until=filters.until),
+    )
     try:
-        statements = list(
-            lrs_client.read(target="/xAPI/statements", query=LRSQuery(query=query))
+        response = Response[VideoViews](
+            status=StatusEnum.SUCCESS, response=indicator.compute()
         )
-    except BackendException as error:
-        raise HTTPException(
-            status_code=status.HTTP_502_BAD_GATEWAY,
-            detail="xAPI statements query failed",
-        ) from error
-
-    date_range = [
-        d.format(settings.DATE_FORMAT)
-        for d in arrow.Arrow.range("day", filters.since, filters.until)
-    ]
-
-    def filter_played_statement(statement) -> bool:
-        """Do not count video played less than the configured time."""
-        return (
-            statement["result"]["extensions"][RESULT_EXTENSION_TIME]
-            < video_settings.VIEWS_COUNT_TIME_THRESHOLD
+        return response
+    except KeyError as exception:
+        logger.error(exception)
+        return Response[Error](
+            status=StatusEnum.FAILED, response=Error(error_message=str(exception))
         )
```

Comment:
Try/except statement scope should be reduced at its smallest portion:

```suggestion
    except KeyError as exception:
        logger.error(exception)
        return Response[Error](
            status=StatusEnum.FAILED, response=Error(error_message=str(exception))
        )
    
    return response
```